dwave_networkx/generators/pegasus.py

```diff
@@ -402,6 +402,88 @@ def defragment_tuple(chimera_coords):
     return defragment_tuple
 
 
+def fragmented_edges(pegasus_graph):
+    """
+    Generator for the edges contained in a Chimera graph obtained by splitting each Pegasus node into
+    six Chimera nodes.  If the Pegasus graph has size parameter m, then the derived graph will be a
+    subgraph of Chimera(6m, 6m, 2) -- that is, a Chimera graph with K2,2 unit tiles.
+
+        The K2,2 Chimera graph uses a coordinate system with an origin at the upper left corner of
+        the graph.
+            y: number of vertical fragments from the top-most row
+            x: number of horizontal fragments from the left-most column
+            u: 1 if it belongs to a horizontal qubit, 0 otherwise
+            r: fragment index on the K2,2 shore
+
+    Parameters
+    ----------
+    pegasus_graph: networkx.graph
+        A pegasus graph
+
+    Returns
+    -------
+    (coord0, coord1), ... : an iterator of tuples
+        Yields the edges contained in the Chimera graph derived from the "fragmentation" construction
+    """
+    offsetlist = [pegasus_graph.graph['vertical_offsets'], pegasus_graph.graph['horizontal_offsets']]
+    offsets = {(u, k): offsetlist[u][k] for u in (0, 1) for k in range(12)}
```

Comment:
I also noticed that `12` is often written out. Is there a possible future where a similar Pegasus structure would be made, except that that the "width" of a Pegasus unit cell would contain more than 12 tracks?

If so, perhaps storing the `12` in a constants file that you can simply import in?

Reply:
Pegasus is never described in any external documentation with anything but 12.  It seems safe to leave this constant.